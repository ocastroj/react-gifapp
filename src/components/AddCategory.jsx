import { useState } from 'react';
import  PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory }) => {

  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({target}) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault();
    const value = inputValue.trim();
    if (value <= 1) return;

    //setCategories(categories => [inputValue, ...categories]);
    onNewCategory(value);
    setInputValue('');
  }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
        <input
          type="text"
          placeholder="Buscar Gifs" 
          value={ inputValue }
          onChange= { onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
