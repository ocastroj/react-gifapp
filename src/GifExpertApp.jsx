import { useState } from 'react';
import { AddCategory, GifGrid } from './components';
// import { AddCategory } from './components/AddCategory';
// import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Cyberpunk']);

  const onAddCategory = ( newCategory ) => {

    if ( categories.includes(newCategory) ) return;
    setCategories( [newCategory, ...categories]);
  }

  return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory 
            //setCategories={ setCategories } 
            onNewCategory={ onAddCategory }
        />

        <button onClick={ onAddCategory }>Agregar</button>

        { categories.map( (category) => (
            <GifGrid key={ category } category={ category }/>
        ))
        }
    </>
  )
}
