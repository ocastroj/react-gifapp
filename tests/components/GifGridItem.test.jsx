import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components";

describe('Prueba <GifGridItem/>', () => { 

    const title = 'One Piece';
    const url= 'https://api.com/q';

    test('Debe de ser igual el snapshot', () => { 
        
        const { container } = render( <GifGridItem title={ title } url={ url }/> );

        expect(container).toMatchSnapshot();
     });

     test('Debe de mostrar la imagen con el URL y ALT indicado', () => { 
        render( <GifGridItem title={ title } url={ url }/> );

        // expect( screen.getByRole('img').src ).toBe(url);
        // expect( screen.getByRole('img').alt ).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
      });

      test('Debe de mostrar el tÃ­tulo en el compnente', () => { 
        render( <GifGridItem title={ title } url={ url }/> );
        expect( screen.getByText( title )).toBeTruthy();
       });
    
 });