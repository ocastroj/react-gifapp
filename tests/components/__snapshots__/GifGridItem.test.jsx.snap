// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifGridItem/> Debe de ser igual el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Piece"
      src="https://api.com/q"
    />
    <p>
      One Piece
       
    </p>
  </div>
</div>
`;
